crates/chain/src/tx_graph.rs

```diff
@@ -1,12 +1,25 @@
 //! Module for structures that store and traverse transactions.
 //!
-//! [`TxGraph`] is a monotone structure that inserts transactions and indexes the spends. The
-//! [`ChangeSet`] structure reports changes of [`TxGraph`] but can also be applied to a
-//! [`TxGraph`] as well. Lastly, [`TxDescendants`] is an [`Iterator`] that traverses descendants of
-//! a given transaction.
+//! [`TxGraph`] is a monotone structure that contains transactions and indexes the spends. It can
+//! contain either full transactions, or partial transactions (transactions for which we have only
+//! certain outputs, which we usually call "floating outputs").
+//!
+//! The graph contains transactions in the form of [`TxNode`] structures. Each node contains the
+//! txid, the transaction (whole or partial), the blocks it's anchored in (see the [`Anchor`]
+//! documentation for more details), and the timestamp of the last time we saw
+//! the transaction as unconfirmed.
 //!
 //! Conflicting transactions are allowed to coexist within a [`TxGraph`]. This is useful for
-//! identifying and traversing conflicts and descendants of a given transaction.
+//! identifying and traversing conflicts and descendants of a given transaction. Some [`TxGraph`]
+//! methods only consider "canonical" (i.e., in the best chain or in mempool) transactions,
+//! we decide which transactions are canonical based on anchors `last_seen_unconfirmed`;
+//! see the [`try_get_chain_position`] documentation for more details.
+//!
+//! The [`ChangeSet`] structure reports changes of [`TxGraph`]; it can be used to either save to persistance,
```

Comment:
persistent storage in better.

In general can we avoid calling things "structures" unless we are really trying to highlight some property of them. This can just be "[`ChangeSet`] reports changes made to to a [`TxGraph`]".